core/src/main/python/wlsdeploy/tool/deploy/deployer_utils.py

```diff
@@ -236,6 +238,39 @@ def ensure_no_uncommitted_changes_or_edit_sessions():
     return
 
 
+def extract_from_uri(model_context, path_value):
+    """
+    If the MBean path attribute has a URI file scheme, look past the scheme and
+    resolve any global tokens.
+
+    :param model_context: current context containing job information
+    :param path_value: MBean attribute path
+    :return: fully resolved URI path, or the path_value if not a URI scheme
+    """
+    uri = URI(path_value)
+    if uri.getScheme() == 'file':
+        return FILE_URI + model_context.replace_token_string(uri.getPath()[1:])
+    return path_value
+
+
+def get_rel_path_from_uri(model_context, path_value):
+    """
+    If the MBean attribute is a URI. To extract it from the archive, need to make
+    it into a relative path.
+    :param model_context: current context containing run information
+    :param path_value: the full value of the path attribute
+    :return: The relative value of the path attribute
+    """
+    uri = URI(path_value)
+    if uri.getScheme() == 'file':
+        value = model_context.tokenize_path(uri.getPath())
+        if value.startswith('@@DOMAIN_HOME@@'):
```

Comment:
model_context has a constant for this, __DOMAIN_HOME_TOKEN, but it would need to become DOMAIN_HOME_TOKEN (without __).  We could make a method in model_context for this, but it seems very specific to this case.

Reply:
I agree. I was totally annoyed that the constant was private and in the model context instance. Seems it used to be somewhere else but could not find it. I suppose I could take this out of deployer_utils and put in model_context, but I did only leave because of its targeted usage. Votes from anyone else?

Reply:
I vote to make it public or a getter if the field needs to be protected for some reason.

Reply:
fixed. made public tokens in model context